src/uu/sort/src/sort.rs

```diff
@@ -266,15 +264,17 @@ type Field = Range<usize>;
 
 #[derive(Clone)]
 pub struct Line {
-    line: String,
+    line: Box<str>,
```

Comment:
`Box<str>` is 2 words long (pointer + len), instead of 3 for String (pointer + len + capacity).

Reply:
Just curious, doesn't using `Box<str>`over `String`makes it slower? 

AFAIK when it has a known size (the capacity), compiler can optimize it pretty efficiently and allocating that string is much faster and makes common operations faster overall.

Reply:
I'm not sure if I'm understanding you correctly and if my understanding here is correct, but I think this should not be slower because with a `Box<str>` we also know the length (it's a fat pointer, essentially the owned version of a `&str`). It's just not possible that the capacity is something other than the length, because we don't store the capacity separately. For the purpose of comparing strings we don't need additional capacity.

I am initially creating a `String`, but convert it using [String::into_boxed_str](https://doc.rust-lang.org/std/string/struct.String.html#method.into_boxed_str). I think, if anything, that conversion could be expensive because we have to re-allocate new memory that has the exact size we need (the documentation says that it drops any excess capacity - not sure if that's what it actually means).